### PR TITLE
Core/ts version task 145 finilize get user order/s endpoints 

### PR DESCRIPTION
**Pull request by franken5tein**
## What does this MR do?

Finalize get user order and orders endpoints and add pagination.

## Check list  

- [x] All Merge Request pipelines should be passed.
- [x] Make sure all the changes are necessary and correct.
- [x] Make sure no needed change missing (all necessary changes are made).
- [x] Merge Request should contains only changes for one purpose (one feature or bug or refactoring).
- [ ] All necessary unit tests are written.
- [x] Critical changes should be tested locally (or on staging).

### CODE REVIEW

**Review by GreatGaryCooper — COMMENTED** [review]
Seems good to me, but please add some test cases using the new pagination feature you just added.
Thanks

**Comment by WilliamHawkingStephen** [comment]
- iam getting internal server error : 

image

**Comment by WilliamHawkingStephen** [comment]
- in think we need to check and validate aboult uint for the page and per_page: 

image

**Comment by WilliamHawkingStephen** [comment]
- we need to add page per_page HTTP validator

**Comment by franken5tein** [comment]
> Seems good to me, but please add some test cases using the new pagination feature you just added. Thanks

Done baby.

File: src/utils/pagination.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        ))
+      : (offset = undefined);
+    return offset;
+  }
+
+  public limit(
+    page: number | undefined,
+    limit: number | undefined,
+  ): number | undefined {
+    return Number(limit);
```
Comment (hypersolo) [inline src/utils/pagination.ts:27]: this returns NaN if limit is undefined. we can simply return limit itself.

File: src/utils/pagination.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  private static instance: Pagination;
+
+  private constructor(private math = Utilities.getInstance().math) {}
+
+  public offset(
+    page: number | undefined,
+    limit: number | undefined,
+  ): number | undefined {
+    let offset: number | undefined;
+    page !== undefined && limit !== undefined
```
Comment (hypersolo) [inline src/utils/pagination.ts:15]: if clause is more readable and easier to debug dude! :love_letter:

File: src/utils/pagination.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  ): number | undefined {
+    return Number(limit);
+  }
+
+  public nextPage(
+    page: number | undefined,
+    per_page: number | undefined,
+    serializedFills: OrderUserSchema[] | FillPublicSchema[],
+  ): number | undefined {
+    return page !== undefined && per_page !== undefined
```
Comment (hypersolo) [inline src/utils/pagination.ts:35]: if clause is more readable and easier to debug dude! :love_letter:

File: src/utils/pagination.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    page: number | undefined,
+    limit: number | undefined,
+  ): number | undefined {
+    return Number(limit);
+  }
+
+  public nextPage(
+    page: number | undefined,
+    per_page: number | undefined,
+    serializedFills: OrderUserSchema[] | FillPublicSchema[],
```
Comment (hypersolo) [inline src/utils/pagination.ts:33]: just passing the length is enough. IMO passing a big object just to understand the length is not ok! :cupid:

File: src/data/database/postgres/repository/fill.adapter.ts [hunk 5]
```diff
@@ -69,14 +69,16 @@ class FillRepoAdapter extends BaseAdapter implements IFillRepo.schema {
...
-        where: whereClause,
-        transaction: this.transaction,
-      })
-      .catch((e) => {
-        throw new DatabaseQueryError(e);
-      });
+    const queryOptions = {
+      where: whereClause,
+      offset: data.offset,
+      limit: data.limit ? data.limit : 1000,
```
Comment (hypersolo) [inline src/data/database/postgres/repository/fill.adapter.ts:75]: what if someone's limit is 5,000,000 !!!
IF limit is not undefined THEN min(limit, 1000) ELSE 1000.

Comment (hypersolo) [reply src/data/database/postgres/repository/fill.adapter.ts:75]: please apply this in pagination class.
remember this affects offset too.

File: src/utils/pagination.ts (hunk 1 above)
Comment (franken5tein) [reply src/utils/pagination.ts]: Done!

File: src/utils/pagination.ts (hunk 2 above)
Comment (franken5tein) [reply src/utils/pagination.ts]: You're right, bro.

File: src/utils/pagination.ts (hunk 4 above)
Comment (franken5tein) [reply src/utils/pagination.ts]: Done!

**Review by hypersolo — APPROVED** [review]

**Review by GreatGaryCooper — APPROVED** [review]
LGTM

**Pull request merged**